neuralmonkey/attention/scaled_dot_product.py

```diff
@@ -0,0 +1,172 @@
+"""The scaled dot-product attention mechanism, defined in Vaswani et al. (2017)
```

Comment:
Myslím, že by si to zasloužilo delší docsting.

Reply:
to je pravda

Reply:
Tak napiš dvěma větama, jak to funguje a už to zmerǧujeme.